### ISSUE TITLE
ReadFromTextLoader(..., supportsSparse, ...) defaults to true?

### ISSUE THREAD

**Issue by stephentoub** [dotnet/machinelearning#2576]
ReadFromTextFile has a `supportsSparse = true` argument.  It seems like this should actually default to false?  Otherwise it tries to interpret all of the input as if it may contain sparse data and fails in the parsing on every line, unless there actually is sparse data.

**Comment by eerhardt** [comment]
@TomFinley @Ivanidzo4ka @yaeldekel - do you know why the current default value for this is `true`? It seems like a very specific format, that the majority of data sets will not conform to.

**Comment by TomFinley** [comment]
I agree with this. The same might be said of "quoting" which fouls people up more often than it helps them. Not only based on your observation from this issue, but if I think back on the years we've had this loader this has been a persistent and confusing problem.

The argument might be made that `TextLoader` and `TextSaver` arguments should be symmetrical, but as I argued in #2452 text saver by default writes out a settings pre-amble anyway which overrides defaults in the text loader, and there are other cases where we have decided that having these be disparate is appropriate. Currently this disparity is limited to header, but if we judge it appropriate there's no reason not to expand it.

If someone knows enough to conform to our format for quoting and sparse vectors, they certainly know enough to set this option, as well as the quoting option.
